src/System.Linq.Parallel/tests/WithCancellationTests.cs

```diff
@@ -157,5 +167,19 @@ public static void DontDoWorkIfTokenAlreadyCanceled()
 
             Assert.NotNull(oce);
         }
+
+        private static void DisposedEnumerator(ParallelQuery<int> query, bool delay = false)
+        {
+            query = query.WithCancellation(new CancellationTokenSource().Token).Select(x => x);
+
+            IEnumerator<int> enumerator = query.GetEnumerator();
+
+            enumerator.MoveNext();
+            if (delay) Task.Delay(1000).Wait();
```

Comment:
>  on my machine the full outerloop currently takes 600+

We really need to bring that down.  10 minutes and something like 300K tests for this library is over the top.  Lots of stuff is being checked and rechecked and rechecked and so on.

> About ~30 seconds

Which is a 5% increase.  That's a lot, for little gain.